pkg/api/norman/customization/gke/handler.go

```diff
@@ -0,0 +1,193 @@
+package gke
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/rancher/norman/api/access"
+	"github.com/rancher/norman/httperror"
+	"github.com/rancher/norman/types"
+	client "github.com/rancher/rancher/pkg/client/generated/management/v3"
+	v1 "github.com/rancher/rancher/pkg/generated/norman/core/v1"
+	"github.com/rancher/rancher/pkg/namespace"
+	"github.com/rancher/rancher/pkg/ref"
+	schema "github.com/rancher/rancher/pkg/schemas/management.cattle.io/v3"
+	"github.com/rancher/rancher/pkg/types/config"
+	"github.com/sirupsen/logrus"
+)
+
+type Capabilities struct {
+	Credentials string `json:"credentials,omitempty"`
+	ProjectID   string `json:"projectId,omitempty"`
+	Zone        string `json:"zone,omitempty"`
+	Region      string `json:"region,omitempty"`
+}
+
+// GKE handler lists available resources in Google API
+type handler struct {
+	Action        string
+	schemas       *types.Schemas
+	secretsLister v1.SecretLister
+}
+
+func NewGKEHandler(scaledContext *config.ScaledContext) http.Handler {
+	return &handler{
+		schemas:       scaledContext.Schemas,
+		secretsLister: scaledContext.Core.Secrets(namespace.GlobalNamespace).Controller().Lister(),
+	}
+}
+
+func (h *handler) ServeHTTP(writer http.ResponseWriter, req *http.Request) {
+
+	writer.Header().Set("Content-Type", "application/json")
+
+	capa := &Capabilities{}
+
+	if credID := req.URL.Query().Get("cloudCredentialId"); credID != "" {
+		if errCode, err := h.getCloudCredential(req, capa, credID); err != nil {
+			handleErr(writer, errCode, err)
+			return
+		}
+	} else if req.Method == http.MethodPost {
+		if errCode, err := h.getCredentialsFromBody(writer, req, capa); err != nil {
+			handleErr(writer, errCode, err)
+			return
+		}
+	} else {
+		handleErr(writer, httperror.Unauthorized.Status, fmt.Errorf("cannot access Google API without credentials to authenticate"))
```

Comment:
I think this is still not backwards compatible, this is what the error was before https://github.com/rancher/rancher/pull/31759/files#diff-d86bb7d4d3eda030b6a6ce369760bcf4f949ea0eda69b9d841f1f741c18a5d69L66

Reply:
Never mind, I guess it wouldn't make sense to return Method Not Allowed any more now that the method isn't checked on line 49. But then this should probably return Bad Request rather than Unauthorized I think.

Reply:
I have changed it to `http.StatusMethodNotAllowed`

Reply:
Looks like its badRequest now